jobs/taar_lite_guidguid.py

```diff
@@ -226,78 +274,68 @@ def load_s3(result_df, date, prefix, bucket):
             value_json[_id] = n
         result_json[key_addon] = value_json
 
-    store_json_to_s3(
-        json.dumps(result_json, indent=2), OUTPUT_BASE_FILENAME, date, prefix, bucket
+    store_json_to_gcs(
+        bucket, prefix, OUTPUT_FILENAME, result_json, date.strftime("%Y%m%d"),
     )
 
 
-def store_json_to_s3(json_data, base_filename, date, prefix, bucket):
-    """Saves the JSON data to a local file and then uploads it to S3.
+def store_json_to_gcs(
+    bucket, prefix, filename, json_obj, iso_date_str, compress=True
+):
+    """Saves the JSON data to a local file and then uploads it to GCS.
 
     Two copies of the file will get uploaded: one with as "<base_filename>.json"
     and the other as "<base_filename><YYYYMMDD>.json" for backup purposes.
 
+    :param bucket: The GCS bucket name.
+    :param prefix: The GCS prefix.
+    :param filename: A string with the base name of the file to use for saving
+        locally and uploading to GCS
     :param json_data: A string with the JSON content to write.
-    :param base_filename: A string with the base name of the file to use for saving
-        locally and uploading to S3.
     :param date: A date string in the "YYYYMMDD" format.
-    :param prefix: The S3 prefix.
-    :param bucket: The S3 bucket name.
     """
-
-    tempdir = tempfile.mkdtemp()
-
-    with selfdestructing_path(tempdir):
-        JSON_FILENAME = "{}.json".format(base_filename)
-        FULL_FILENAME = os.path.join(tempdir, JSON_FILENAME)
-        with open(FULL_FILENAME, "w+") as json_file:
-            json_file.write(json_data)
-
-        archived_file_copy = "{}{}.json".format(base_filename, date)
-
-        # Store a copy of the current JSON with datestamp.
-        write_to_s3(FULL_FILENAME, archived_file_copy, prefix, bucket)
-        write_to_s3(FULL_FILENAME, JSON_FILENAME, prefix, bucket)
-
-
-def write_to_s3(source_file_name, s3_dest_file_name, s3_prefix, bucket):
-    """Store the new json file containing current top addons per locale to S3.
-
-    :param source_file_name: The name of the local source file.
-    :param s3_dest_file_name: The name of the destination file on S3.
-    :param s3_prefix: The S3 prefix in the bucket.
-    :param bucket: The S3 bucket.
-    """
-    client = boto3.client(
-        service_name="s3", region_name="us-west-2", **aws_env_credentials()
-    )
-    transfer = boto3.s3.transfer.S3Transfer(client)
-
-    # Update the state in the analysis bucket.
-    key_path = s3_prefix + s3_dest_file_name
-    transfer.upload_file(source_file_name, bucket, key_path)
-
-
-def load_amo_external_whitelist():
+    byte_data = json.dumps(json_obj).encode("utf8")
+
+    byte_data = bz2.compress(byte_data)
+    logger.info(f"Compressed data is {len(byte_data)} bytes")
+
+    client = storage.Client()
```

Comment:
i generally like to wrap any google client code in a try/catch for exception/error logging, but this is upto you

Reply:
good idea, added